### ISSUE TITLE
Re-add `etcd` node error when using rke up command

### ISSUE THREAD

**Issue by Jason-ZW** [rancher/rke#704]
**RKE version:**
github.com/rancher/rke v0.1.8-rc2
**Docker version: (`docker version`,`docker info` preferred)**
docker-17.03.2-ce
**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
RancherOS-v1.4.0
**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
AWS
**cluster.yml file:**
```
- address: 172.31.2.71
  port: ""
  internal_address: ""
  role:
  - controlplane
  - worker
  - etcd
  hostname_override: ""
  user: rancher
  docker_socket: ""
  ssh_key: ""
  ssh_key_path: /home/rancher/.ssh/id_rsa
  labels: {}
- address: 172.31.8.170
  port: ""
  internal_address: ""
  role:
  - controlplane
  - worker
  - etcd
  hostname_override: ""
  user: rancher
  docker_socket: ""
  ssh_key: ""
  ssh_key_path: /home/rancher/.ssh/id_rsa
  labels: {}
services:
  etcd:
    image: ""
    extra_args: {}
    extra_binds: []
    extra_env: []
    external_urls: []
    ca_cert: ""
    cert: ""
    key: ""
    path: ""
    snapshot: false
    retention: ""
    creation: ""
  kube-api:
    image: ""
    extra_args:
      feature-gates: MountPropagation=true
    extra_binds: []
    extra_env: []
    service_cluster_ip_range: ""
    service_node_port_range: ""
    pod_security_policy: false
  kube-controller:
    image: ""
    extra_args: {}
    extra_binds: []
    extra_env: []
    cluster_cidr: ""
    service_cluster_ip_range: ""
  scheduler:
    image: ""
    extra_args: {}
    extra_binds: []
    extra_env: []
  kubelet:
    image: ""
    extra_args:
      feature-gates: MountPropagation=true
      volume-plugin-dir: /opt/rke/var/lib/kubelet/volumeplugins/
    extra_binds: []
    extra_env: []
    cluster_domain: ""
    infra_container_image: ""
    cluster_dns_server: ""
    fail_swap_on: false
  kubeproxy:
    image: ""
    extra_args: {}
    extra_binds: []
    extra_env: []
network:
  plugin: flannel
  options: {}
authentication:
  strategy: ""
  options: {}
  sans: []
addons: ""
addons_include: []
system_images:
  etcd: ""
  alpine: ""
  nginx_proxy: ""
  cert_downloader: ""
  kubernetes_services_sidecar: ""
  kubedns: ""
  dnsmasq: ""
  kubedns_sidecar: ""
  kubedns_autoscaler: ""
  kubernetes: ""
  flannel: ""
  flannel_cni: ""
  calico_node: ""
  calico_cni: ""
  calico_controllers: ""
  calico_ctl: ""
  canal_node: ""
  canal_cni: ""
  canal_flannel: ""
  wave_node: ""
  weave_cni: ""
  pod_infra_container: ""
  ingress: ""
  ingress_backend: ""
ssh_key_path: ""
ssh_agent_auth: false
authorization:
  mode: ""
  options: {}
ignore_docker_version: false
kubernetes_version: v1.10.3-rancher2-1
private_registries: []
ingress:
  provider: ""
  options: {}
  node_selector: {}
  extra_args: {}
cluster_name: ""
cloud_provider:
  name: ""
prefix_path: ""
addon_job_timeout: 0
bastion_host:
  address: ""
  port: ""
  user: ""
  ssh_key: ""
  ssh_key_path: ""
```
**Steps to Reproduce:**
After `rke up` cluster, I remove `172.31.8.170` from `cluster.yml` then re-execute `rke up` command. The error occur:
```
2018-06-19 08:24:50.516845 I | rafthttp: peer 98c792dd82050a1a became active
2018-06-19 08:24:50.523881 W | rafthttp: rejected the stream from peer 65b6768f402d213b since it was removed
2018-06-19 08:24:50.603540 E | rafthttp: failed to dial 98c792dd82050a1a on stream Message (unhandled error "to field mismatch\n" when precondition failed)
2018-06-19 08:24:50.603592 I | rafthttp: peer 98c792dd82050a1a became inactive
2018-06-19 08:24:50.630566 W | rafthttp: rejected the stream from peer 65b6768f402d213b since it was removed
2018-06-19 08:24:50.637995 W | rafthttp: rejected the stream from peer 65b6768f402d213b since it was removed
2018-06-19 08:24:50.745255 W | rafthttp: rejected the stream from peer 65b6768f402d213b since it was removed
2018-06-19 08:24:50.750785 W | rafthttp: rejected the stream from peer 65b6768f402d213b since it was removed
``` 
I’m actually working on the rancher cube project, where I need to use the rke command add/rm rke node(The node is ros instance).Everything is normal except adding and removing etcds. I must remove the directory `/opt/rke/var/lib/etcd/*` on etcd node by manual, but the scenario better to automatically.

**Comment by galal-hussein** [comment]
@Jason-ZW I tested etcd add/remove and it seems to work okay, after removing the node from cluter.yml and re-execute rke up again, what are the rke logs, also can you `etcdctl cluster-health`

**Comment by Jason-ZW** [comment]
@galal-hussein 
RKE logs:
```
INFO[0000] Building Kubernetes cluster
INFO[0000] [dialer] Setup tunnel for host [172.31.2.71]
INFO[0000] [dialer] Setup tunnel for host [172.31.8.170]
INFO[0000] [state] Found local kube config file, trying to get state from cluster
INFO[0000] [state] Fetching cluster state from Kubernetes
INFO[0000] [state] Successfully Fetched cluster state to Kubernetes ConfigMap: cluster-state
INFO[0000] [certificates] Getting Cluster certificates from Kubernetes
INFO[0000] [certificates] Successfully fetched Cluster certificates from Kubernetes
INFO[0000] [network] Deploying port listener containers
INFO[0000] [network] Successfully started [rke-etcd-port-listener] container on host [172.31.8.170]
INFO[0000] [network] Successfully started [rke-cp-port-listener] container on host [172.31.8.170]
INFO[0000] [network] Successfully started [rke-worker-port-listener] container on host [172.31.8.170]
INFO[0000] [network] Port listener containers deployed successfully
INFO[0000] [network] Running etcd <-> etcd port checks
INFO[0001] [network] Successfully started [rke-port-checker] container on host [172.31.8.170]
INFO[0001] [network] Successfully started [rke-port-checker] container on host [172.31.2.71]
INFO[0001] [network] Running control plane -> etcd port checks
INFO[0001] [network] Successfully started [rke-port-checker] container on host [172.31.8.170]
INFO[0001] [network] Successfully started [rke-port-checker] container on host [172.31.2.71]
INFO[0001] [network] Running control plane -> worker port checks
INFO[0001] [network] Successfully started [rke-port-checker] container on host [172.31.8.170]
INFO[0001] [network] Successfully started [rke-port-checker] container on host [172.31.2.71]
INFO[0002] [network] Running workers -> control plane port checks
INFO[0002] [network] Successfully started [rke-port-checker] container on host [172.31.2.71]
INFO[0002] [network] Successfully started [rke-port-checker] container on host [172.31.8.170]
INFO[0002] [network] Checking KubeAPI port Control Plane hosts
INFO[0002] [network] Removing port listener containers
INFO[0002] [remove/rke-etcd-port-listener] Successfully removed container on host [172.31.8.170]
INFO[0002] [remove/rke-cp-port-listener] Successfully removed container on host [172.31.8.170]
INFO[0002] [remove/rke-worker-port-listener] Successfully removed container on host [172.31.8.170]
INFO[0002] [network] Port listener containers removed successfully
INFO[0002] [reconcile] Reconciling cluster state
INFO[0002] [reconcile] Check etcd hosts to be deleted
INFO[0002] [reconcile] Check etcd hosts to be added
INFO[0002] [certificates] Regenerating new etcd-172.31.8.170 certificate and key
INFO[0002] [certificates] Successfully generated new etcd-172.31.8.170 certificate and key
INFO[0008] [add/etcd] Adding member [etcd-172.31.8.170] to etcd cluster
INFO[0008] [add/etcd] Successfully Added member [etcd-172.31.8.170] to etcd cluster
INFO[0008] [etcd] Successfully updated [etcd] container on host [172.31.2.71]
INFO[0008] [etcd] Successfully started [rke-log-linker] container on host [172.31.2.71]
INFO[0009] [remove/rke-log-linker] Successfully removed container on host [172.31.2.71]
INFO[0009] [etcd] Successfully started [etcd] container on host [172.31.8.170]
INFO[0009] [etcd] Successfully started [rke-log-linker] container on host [172.31.8.170]
INFO[0009] [remove/rke-log-linker] Successfully removed container on host [172.31.8.170]
FATA[0101] Failed to reconcile etcd plane: [etcd] Etcd Cluster is not healthy
```

**Comment by deniseschannon** [comment]
Available with v0.1.8-rc6

**Comment by Jason-ZW** [comment]
Tested. LGTM
